From: Jinoh Kang <jinoh.kang.kr@gmail.com>
Subject: [PATCH resend] wineboot: Don't use ai_canonname for ComputerName if it is "localhost".
Message-Id: <f4c3380d-9691-95a5-de93-049e06302d7e@gmail.com>
Date: Sat, 5 Mar 2022 23:58:58 +0900

Signed-off-by: Jinoh Kang <jinoh.kang.kr@gmail.com>
---
Some Linux distros resolves the current hostname back to "localhost",
so it needs to be explicitly filtered out.

 programs/wineboot/wineboot.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/programs/wineboot/wineboot.c b/programs/wineboot/wineboot.c
index 11111111111..11111111111 100644
--- a/programs/wineboot/wineboot.c
+++ b/programs/wineboot/wineboot.c
@@ -1337,7 +1337,9 @@ static void create_computer_name_keys(void)
 
     if (gethostname( buffer, sizeof(buffer) )) return;
     hints.ai_flags = AI_CANONNAME;
-    if (!getaddrinfo( buffer, NULL, &hints, &res )) name = res->ai_canonname;
+    if (!getaddrinfo( buffer, NULL, &hints, &res ) &&
+        res->ai_canonname && strcasecmp(res->ai_canonname, "localhost") != 0)
+        name = res->ai_canonname;
     dot = strchr( name, '.' );
     if (dot) *dot++ = 0;
     else dot = name + strlen(name);

-- 
2.35.1

